File: cpp/2248.intersection-of-multiple-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<vector<int>>& nums) {
	    unordered_map<int, int> freq;
	    for(auto &row:nums){
		    for(auto num:row)freq[num]++;
	    }
	    vector<int> res;
	    for(auto [num, rep]:freq){
		    if(rep == nums.size()) res.push_back(num);
	    }
	    sort(res.begin(), res.end());
	    return res;
    }
};

File: python/2248.intersection-of-multiple-arrays.py
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        freq = {}
        for row in nums:
            for num in row:
                freq[num] = freq.get(num, 0)+1
        res = [ key  for key,val in freq.items() if val == len(nums)]
        res.sort()
        return res
